net: shell: TCP stack agnostic shell

loicpoulain opened the pull request:
Now that TCP2 is the default stack, make the shell compatible
with both stacks.

Note: net_tcp_foreach is not implemented in TCP2, so related
code depends on TCP1.

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

jukkar commented on samples/net/wifi/prj.conf:
```diff
@@ -28,3 +28,4 @@ CONFIG_NET_STATISTICS_PERIODIC_OUTPUT=n
 CONFIG_WIFI=y
 CONFIG_WIFI_LOG_LEVEL_ERR=y
 CONFIG_NET_L2_WIFI_SHELL=y
+
```
Unncessary empty line here, can be removed

Commit:
net: shell: TCP stack agnostic shell

Now that TCP2 is the default stack, make the shell compatible
with both stacks.

Note: net_tcp_foreach is not implemented in TCP2, so related
code depends on TCP1.

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>
